data/coerce/compound.go

```diff
@@ -189,7 +189,7 @@ func ToArrayIfNecessary(val interface{}) (interface{}, error) {
 		if t != "" {
 			err := json.Unmarshal([]byte(t), &a)
 			if err != nil {
-				return nil, fmt.Errorf("unable to coerce %#v to map[string]interface{}", val)
+				return t, nil
```

Comment:
@fm-tibco Should I just return the current string or should I return the array ?